CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+
+- A concise and descriptive title.
+
+- Steps to reproduce the issue along with and explanation of the expected behavior.
+
+- Relevant details, such as screenshots, logs, or error messages and an environment description in the following form:
+
+    ```plaintext
+    - OS version: Ubuntu 22.04.5 LTS
+    - OpenFOAM version: openfoam2312
+    - solids4foam version: 2.1
+    - Docker image (if applicable)
+    ```
+
+### 2. Suggesting Features
+
+We welcome ideas for new features! To suggest a feature:
+
+- Open a [GitHub issue](https://github.com/solids4foam/solids4foam/issues) with title: **Feature request: feature title**.
+- Provide a detailed description of the proposed feature.
+- Explain its potential benefits for the solids4foam toolbox.
+
+### 3. Submitting Code Changes
+
+#### Step 1: Fork and Clone
+
+1. Fork the repository by clicking the "Fork" button on GitHub.
+
+2. Clone your fork locally:
+
+    ```bash
+    git clone https://github.com/your-username/solids4foam.git
+    cd solids4foam
+    ```
+
+#### Step 2: Create a Branch
+
+Create a new branch for your changes:
+
+```bash
+git checkout -b branch_name
+```
+
+Branch naming conventions:
+
+- **New features**: Start with `feature-name` (e.g., `feature-newSolver`).
+- **Bug fixes**: Start with `fix-name` (e.g., `fix-boundaryConditions`).
+- **Code refactoring**: Start with `refactor-name` (e.g., `refactor-meshHandling`).
+- **Documentation changes**: Start with `docs-name` (e.g., `docs-update-readme` or `docs-fix-typos`).
+
+#### Step 3: Make Your Changes
+
+- Follow the [OpenFOAM Coding Style Guide](https://openfoam.org/dev/coding-style-guide/)).
+- Ensure that your code can be compiled and adheres the existing code structure and organization.
+- Test your changes thoroughly with relevant test cases. Include or update tutorial cases if applicable.
+- Document your code with clear comments.
+- Before submitting, check the code using available linters or style checkers ([shellcheck](https://www.shellcheck.net/#), [markdownlint](https://dlaa.me/markdownlint/)).
+
+#### Step 4: Commit Your Changes
+
+Write clear and descriptive commit messages and commit your changes. Reference any related issues using hashtags if applicable::
+
+```bash
+git add file_name.C
+git commit -m "Fix for issue #51"
+```
+
+#### Step 5: Push and Open a Pull Request
+
+Push your branch to your fork:
+
+```bash
+git push origin branch_name
+```
+
+Go to the original solids4foam repository on GitHub and open a pull request (target the development branch for your pull request). Include:
+
+- **A clear title**, such as `Fix: Issue #123` or `Feature: Add new functionality`.
+- **A detailed description** of your changes.
+- Any **related issue numbers** (e.g., `Closes #123`).
+- **Instructions or example cases** to test your contribution, if applicable.
+
+## Questions?
+
+If you have any questions or need assistance, feel free to open a [discussion](https://github.com/solids4foam/solids4foam/discussions).
```

Comment:
You could somewhere clarify what kind of discussions you expect in the GitHub Discussions (probably interesting for developers) and what in CFDOnline (probably user support).

Reply:
Resolved in bd6e54f9.

Thanks!